src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py

```diff
@@ -869,30 +836,10 @@ def cli_cosmosdb_network_rule_remove(cmd,
     if not rule_removed:
         raise CLIError("This rule does not exist for the Cosmos DB account")
 
-    locations = []
```

Comment:
Why do you remove location related code here?

Reply:
This command is specifically to update virtual_network_rules on the account. All the other code I removed was just copying the existing account state so that other properties are not incorrectly update or reset to defaults. Now that we are using "patch", we only need to include the relevant properties since empty/null values are ignored by the RP.